ballerina/llm_client_azure_open_ai.bal

```diff
@@ -41,23 +41,26 @@ isolated distinct client class AzureOpenAIModel {
 
         http:BearerTokenConfig|ApiKeysConfig auth = connectionConfig.auth;
         self.headers = auth is ApiKeysConfig ? {"api-key": auth?.apiKey} : {};
-        
+
         self.cl = check new (azureOpenAIModelConfig.serviceUrl, httpClientConfig);
 
         self.deploymentId = deploymentId;
         self.apiVersion = apiVersion;
     }
 
-    isolated remote function chat(AzureOpenAICreateChatCompletionRequest chatBody) 
+    isolated remote function chat(AzureOpenAICreateChatCompletionRequest chatBody)
             returns AzureOpenAICreateChatCompletionResponse|error {
         string resourcePath = string `/deployments/${check getEncodedUri(self.deploymentId)}/chat/completions`;
         resourcePath = string `${resourcePath}?${check getEncodedUri("api-version")}=${self.apiVersion}`;
         return self.cl->post(resourcePath, chatBody, self.headers);
     }
 
     isolated remote function call(Prompt prompt, typedesc<anydata> expectedResponseTypedesc) returns anydata|error {
+        SchemaResponse schemaResponse = getExpectedResponseSchema(expectedResponseTypedesc);
         AzureOpenAICreateChatCompletionRequest chatBody = {
-            messages: [{role: "user", content: getPromptWithExpectedResponseSchema(prompt, expectedResponseTypedesc)}]
+            messages: [{role: "user", content: buildPromptString(prompt)}],
+            tools: getTools(schemaResponse.schema),
```

Comment:
Rename `getTools` to be more specific. We know exactly what kind of tool we are using for what here.

Reply:
Will rename as `getToolsForGenerateTheLlmResult`